src/odb/include/odb/db.h

```diff
@@ -7504,31 +7563,90 @@ class dbModInst : public dbObject
   static dbSet<dbModInst>::iterator destroy(dbSet<dbModInst>::iterator& itr);
 
   static dbModInst* getModInst(dbBlock* block_, uint dbid_);
+  dbModITerm* getdbModITerm(dbId<dbModITerm>);
 
-  std::string getName() const;
+  // std::string getName() const;
 
   std::string getHierarchicalName() const;
+
+  bool getPinAtIx(unsigned ix, dbModITerm*& ret) const;
+  bool findModITerm(const char* name, dbModITerm*& ret) const;
+
+  // Todo make this a non stl vector.
+  std::vector<dbId<dbModITerm>>& getPinVec() const;
+  std::vector<dbId<dbModITerm>>::iterator getPinVecBegin() const;
+  std::vector<dbId<dbModITerm>>::iterator getPinVecEnd() const;
```

Comment:
Why are getPinVecBegin & getPinVecEnd needed if you have getPinVec ?  It would be best for getPinVec() to return a const reference

Reply:
Done -- in all cases (there were a few others) am not just passing back the constant reference to whole vector and removed the unnecessary begin/end markers.